src/idpi/operators/time_operators.py

```diff
@@ -34,23 +52,149 @@ def delta(field: xr.DataArray, dtime: np.timedelta64) -> xr.DataArray:
     Raises
     ------
     ValueError
-        if dtime is not multiple of the field time step.
+        if dtime is not multiple of the field time step
+        or if the time step is not regular.
 
     Returns
     -------
     xr.DataArray
-        The Field difference for the given time delta.
+        The field difference for the given time delta.
 
     """
-    coord = field.valid_time
-    condition = coord - coord[0] == dtime
-
-    try:
-        [index] = np.nonzero(condition.values)
-    except ValueError:
-        msg = "Provided dtime is not a multiple of the time step."
-        raise ValueError(msg)
-
-    result = field - field.shift(time=index.item())
+    nsteps = _nsteps(field.valid_time, dtime)
+    result = field - field.shift(time=nsteps)
     result.attrs = field.attrs
     return result
+
+
+def min(field: xr.DataArray, dtime: np.timedelta64) -> xr.DataArray:
+    """Compute minimum aggregate for a given delta in time.
+
+    Parameters
+    ----------
+    field : xr.DataArray
+        Field that contains the input data.
+    dtime : np.timedelta64
+        Time delta for which to evaluate the minimum.
+
+    Raises
+    ------
+    ValueError
+        if dtime is not multiple of the field time step
+        or if the time step is not regular.
+
+    Returns
+    -------
+    xr.DataArray
+        The field minimum for the given time delta.
+
+    """
+    nsteps = _nsteps(field.valid_time, dtime)
+    return field.rolling(time=nsteps).min()
+
+
+def max(field: xr.DataArray, dtime: np.timedelta64) -> xr.DataArray:
+    """Compute maximum aggregate for a given delta in time.
+
+    Parameters
+    ----------
+    field : xr.DataArray
+        Field that contains the input data.
+    dtime : np.timedelta64
+        Time delta for which to evaluate the maximum.
+
+    Raises
+    ------
+    ValueError
+        if dtime is not multiple of the field time step
+        or if the time step is not regular.
+
+    Returns
+    -------
+    xr.DataArray
+        The field maximum for the given time delta.
+
+    """
+    nsteps = _nsteps(field.valid_time, dtime)
+    return field.rolling(time=nsteps).max()
+
+
+def avg(field: xr.DataArray, dtime: np.timedelta64) -> xr.DataArray:
```

Comment:
In order to distinguish this operator from the regular avg, min, max, and having to overload, I was thinking that moving average, or rolling average is a more precise function name. Also dtime -> dtime_window would help understand the behaviour. 

For example: 
```
rolling_avg(field: xr.Darray, dtime_window: np.timedelta64)
```

Reply:
alternatively, I wonder the value of these functions, since I think 
```
field.rolling(time=nsteps).max()
```
is the most readable version and does not require writing multiple function for the various operators min, max, etc. 

Therefore I would suggest an alternative, making nsteps a public function, which can be used directly in the products before the rolling function.